File: app/src/main/java/com/sevtinge/cemiuiler/module/securitycenter/sidebar/video/DisableRemoveScreenHoldOn.java
package com.sevtinge.cemiuiler.module.securitycenter.sidebar.video;

import com.sevtinge.cemiuiler.module.base.BaseHook;
import com.sevtinge.cemiuiler.module.securitycenter.SecurityCenterDexKit;

import java.lang.reflect.Method;
import java.util.List;
import java.util.Objects;

import de.robv.android.xposed.XC_MethodReplacement;
import de.robv.android.xposed.XposedBridge;
import io.luckypray.dexkit.descriptor.member.DexMethodDescriptor;

public class DisableRemoveScreenHoldOn extends BaseHook {
    @Override
    public void init() {
        try {
            List<DexMethodDescriptor> result = Objects.requireNonNull(SecurityCenterDexKit.mSecurityCenterResultMap.get("RemoveScreenHoldOn"));
            for (DexMethodDescriptor descriptor : result) {
                Method removeScreenHoldOn = descriptor.getMethodInstance(lpparam.classLoader);
                log("removeScreenHoldOn method is " + removeScreenHoldOn);
                if (removeScreenHoldOn.getReturnType() == boolean.class) {
                    XposedBridge.hookMethod(removeScreenHoldOn, XC_MethodReplacement.returnConstant(false));
                }
            }
        } catch (Throwable e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/sevtinge/cemiuiler/module/hook/securitycenter/sidebar/video/DisableRemoveScreenHoldOn.kt
package com.sevtinge.cemiuiler.module.hook.securitycenter.sidebar.video

import com.github.kyuubiran.ezxhelper.EzXHelper
import com.github.kyuubiran.ezxhelper.HookFactory.`-Static`.createHook
import com.sevtinge.cemiuiler.module.base.BaseHook
import com.sevtinge.cemiuiler.utils.DexKit.dexKitBridge

object DisableRemoveScreenHoldOn : BaseHook() {
    /*override fun init() {
        try {
            val result: List<DexMethodDescriptor> =
                Objects.requireNonNull(SecurityCenterDexKit.mSecurityCenterResultMap.get("RemoveScreenHoldOn"))
            for (descriptor in result) {
                val removeScreenHoldOn: Method = descriptor.getMethodInstance(lpparam.classLoader)
                log("removeScreenHoldOn method is $removeScreenHoldOn")
                if (removeScreenHoldOn.returnType == Boolean::class.javaPrimitiveType) {
                    XposedBridge.hookMethod(
                        removeScreenHoldOn,
                        XC_MethodReplacement.returnConstant(false)
                    )
                }
            }
        } catch (e: Throwable) {
            e.printStackTrace()
        }
    }*/

    private val screen by lazy {
        dexKitBridge.findMethod {
            matcher {
                usingStrings = listOf("remove_screen_off_hold_on")
                returnType = "boolean"
            }
        }.firstOrNull()?.getMethodInstance(EzXHelper.classLoader)
    }

    override fun init() {
        screen?.createHook {
            before {
                it.result = false
            }
        }
    }
}
